### PR TITLE
automountServiceAccountToken documentation

### PR DESCRIPTION
**Pull request by liggitt**
doc for https://github.com/kubernetes/kubernetes/pull/37953

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/2769)

### CODE REVIEW

**Comment by liggitt** [comment]
@ericchiang PTAL

File: docs/user-guide/service-accounts.md [hunk 1]
```diff
@@ -25,15 +25,42 @@ When they do, they are authenticated as a particular Service Account (e.g.
...
-you get the raw json or yaml for a pod you have created (e.g. `kubectl get
-pods/podname -o yaml`), you can see the `spec.serviceAccount` field has been
+When you create a pod, if you do not specify a service account, it is
+automatically assigned the `default` service account in the same namespace.
+If you get the raw json or yaml for a pod you have created (e.g. `kubectl get pods/podname -o yaml`),
+you can see the `spec.serviceAccountName` field has been
 [automatically set](/docs/user-guide/working-with-resources/#resources-are-automatically-modified).
 
-With service accounts, you can access the API inside the pod using a proxy or with a client library,
+You can access the API from inside a pod using automatically mounted service account credentials,
```
Comment (ericchiang) [inline docs/user-guide/service-accounts.md]: Maybe note that in cluster that enable authorization, service accounts have no permissions when accessing the API and must explicitly be granted them? It's mentioned a little later, but might help keep someone from freaking out here.

Comment (liggitt) [inline docs/user-guide/service-accounts.md]: noted

**Comment by ericchiang** [comment]
lgtm

**Comment by chenopis** [comment]
@liggitt Sorry, we've been in the process of migrating the content into a new structure, so `/docs/user-guide/service-accounts.md` was moved to `/docs/tasks/configure-pod-container/configure-service-account.md`. If you could apply the changes there, that should resolve the merge conflict, then I can merge this PR. Thx

**Comment by liggitt** [comment]
@chenopis done

**Comment by chenopis** [comment]
@liggitt Thanks!

**Pull request merged**